### ISSUE TITLE
[Multi-device] Issues with ```lax.pmean()``` (parent issue)

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-mlir#3645]
### Describe the bug

Context for this set of issues are efforts in bringing up data-parallel MNIST training.
```lax.pmean()``` appears during loss calculation (averaging local losses from devices) and during parameters update, when all local gradients from devices are averaged and then applied back to parameters on each device.

Possible workaround for ```lax.pmean()``` could be ```lax.all_gather() - > jnp.mean()``` however this isn't functional yet, given the issue https://github.com/tenstorrent/tt-metal/issues/22654
